datafusion/physical-plan/src/joins/stream_join_utils.rs

```diff
@@ -740,20 +617,338 @@ pub fn record_visited_indices<T: ArrowPrimitiveType>(
     }
 }
 
+#[macro_export]
+macro_rules! handle_state {
```

Comment:
I think it would help to add some comments here explaining what this macro was used for

Reply:
Sure, I will.